File: src/main/java/com/gaye/dotoApp/dto/AdminRequest.java
package com.gaye.dotoApp.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;
import jakarta.validation.constraints.NotNull;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@AllArgsConstructor
@NoArgsConstructor
@Data
public class AdminRequest {

    @NotNull(message = "Le prenom est requis")
    @NotBlank(message = "Le prenom n'est pas valide")
    private String firstName;

    @NotNull(message = "Le nom est requis")
    @NotBlank(message = "Le nom n'est pas valide")
    private String lastName;

    @NotNull(message = "Le nom est requis")
    @NotBlank(message = "Le nom n'est pas valide")
    @Email(message = "L'email n'est pas valide")
    private String email;

    @NotNull(message = "Le mot de passe est requis")
    @NotBlank(message = "Le mot de passe n'est pas valide")
    private String password;
}

File: src/services/types/models/AdminRequest.ts
// AdminRequest.ts
// ✅ Type représentant les données envoyées au backend pour créer ou mettre à jour un administrateur

export interface AdminRequest {
  firstName: string; // Prénom de l'administrateur
  lastName: string; // Nom de famille de l'administrateur
  email: string; // Adresse email de l'administrateur
  password: string; // Mot de passe de l'administrateur
}
